### DISCUSSION TITLE
Support shadow Dom as root node

### DISCUSSION THREAD

**Discussion by zhuyifang** [quasarframework/quasar#13575] [Ideas / Proposals]
quasar awesome Vue component
I have developed numerous applications using it.
I have a few suggestions:
1. The mounted root node can be configured through quasar.config.js
2. The root node supports shadow-root
3. Components such as dialog should not be rendered in the body, but limited to the configured root node

At present, I have achieved the above 1 and 2 points through simple modifications;

Why?

I am using quasar to develop a chrome plug-in. One of the requirements needs to insert some content into the current web page, and I want to develop this content through quasar. If I use the traditional method, many of the global styles in the compiled output index.xxxxx.css will The original web page was polluted

**Comment by chan-dev** [comment]
Where you able to achieve item 3? I also do have this problem where the modal is injected in the body instead of inside the shadow root of the main component.

**Reply by zhuyifang** [reply]
我在项目中引入quasar编译后的js，也就是umd，然后将源代码中 的xxx.body 改成：shadow
